testsuite/MDAnalysisTests/core/test_topologyattrs.py

```diff
@@ -494,3 +495,45 @@ def test_set_crosslevel(self):
                     # but will fail to be set
                     continue
                 yield self._check_crosslevel_fail, component, singular_attr
+
+
+class TestInstantSelectorDeprecation(object):
+    """Test the deprecation warnings for instant selectors
+
+    Instant selectors are deprecated since version 0.16.2. PR #1403 introduced
+    deprecation warnings for these selectors.
+    """
+    @staticmethod
+    @pytest.fixture()
+    def universe():
+        return mda.Universe(PSF, DCD)
+
+    @pytest.mark.parametrize('instruction', (
+        'universe.atoms.CA',
+        'universe.residues.LYS',
+        'universe.segments.s4AKE',
+        pytest.param('universe.s4AKE',
+                     marks=pytest.mark.xfail(reason="Issue #1478")),
+    ))
+    def test_deprecation(self, universe, instruction):
+        """Test that the warnings are issued when required.
+        """
+        with pytest.deprecated_call():
+            exec(instruction)
```

Comment:
I was going to add a comment for pylint to ignore the `exec` here. Using `exec` in this specific context makes both the test, and the verbose output of the test more readable. The warning about `exec` comes from the security risk that exists when `exec` is used on untrusted input; here the input is safe and cannot be forged to take advantage of the code.